### PR TITLE
Set selinux to permissive mode for installers

### PR DESCRIPTION
**Pull request by achilleas-k**
This PR sets selinux to permissive mode in all installers in RHEL 8.6 and RHEL 9.0 (see [rhbz#2031939](https://bugzilla.redhat.com/show_bug.cgi?id=2031939)).

I also added configs to the `format-request-map` to be able to generate test manifests for `edge-installer` and `edge-simplified-installer` and updated the Go test to use the ostree parameters from the compose request in the test file to verify.

### CODE REVIEW

**Comment by teg** [comment]
I could not find where we do the selinux labelling of the installer trees... How does this work correctly if the trees are unlabelled and selinux is enforcing?

**Review by thozza — DISMISSED** [review]
Everything looks good. Thanks for adding the configs to `format-request-map`. I was adding it as well in one PR that I have in progress, but didn't do the necessary changes to pass OSTree parameters. So I'm glad that you did it 🙂

**Comment by thozza** [comment]
> I could not find where we do the selinux labelling of the installer trees... How does this work correctly if the trees are unlabelled and selinux is enforcing?

My gut feeling is that we need to have the selinux labeling stage only in pipelines which are creating some configuration files to ensure that those are correctly labeled. But I'm not 100% sure...

**Comment by teg** [comment]
> > I could not find where we do the selinux labelling of the installer trees... How does this work correctly if the trees are unlabelled and selinux is enforcing?
> 
> My gut feeling is that we need to have the selinux labeling stage only in pipelines which are creating some configuration files to ensure that those are correctly labeled. But I'm not 100% sure...

Ah, I thought we completely disabled SELinux from the rpm stage so you'd end up with a completely unlabelled tree if you don't relabel. But I could very well be wrong on that.

**Comment by thozza** [comment]
> > > I could not find where we do the selinux labelling of the installer trees... How does this work correctly if the trees are unlabelled and selinux is enforcing?
> > 
> > 
> > My gut feeling is that we need to have the selinux labeling stage only in pipelines which are creating some configuration files to ensure that those are correctly labeled. But I'm not 100% sure...
> 
> Ah, I thought we completely disabled SELinux from the rpm stage so you'd end up with a completely unlabelled tree if you don't relabel. But I could very well be wrong on that.

I don't see any mentions of selinux in the osbuild rpm stage. And this comment in the selinux stage supports my assumption - https://github.com/osbuild/osbuild/blob/97905f6e0c27ab968694e1771a94249143a18b0a/stages/org.osbuild.selinux#L17-L19

**Comment by achilleas-k** [comment]
Dropped the change from `edge-simplified-installer`

**Review by teg — DISMISSED** [review]

File: internal/distro/rhel90/pipelines.go [hunk 1]
```diff
@@ -805,6 +805,7 @@ func simplifiedInstallerTreePipeline(repos []rpmmd.RepoConfig, packages []rpmmd.
 		dracutStageOptions.Install = []string{"/fdo_diun_pub_key_root_certs.pem"}
 	}
 	p.AddStage(osbuild.NewDracutStage(dracutStageOptions))
+
```
Comment (gicmo) [inline internal/distro/rhel90/pipelines.go:808]: Omg, hope David doesn't see that!

**Review by gicmo — COMMENTED** [review]
Is `permissive` also what the standard installer uses? I was thinking if the concern is AVC messages in the logs, wouldn't they also happen for `permissive`?

File: internal/distro/rhel90/pipelines.go (hunk 1 above)
Comment (achilleas-k) [reply internal/distro/rhel90/pipelines.go]: See what? >_>

Comment (gicmo) [reply internal/distro/rhel90/pipelines.go]: sneaky!

**Comment by achilleas-k** [comment]
> Is `permissive` also what the standard installer uses? 

Yes (according to linked bug and my own verification).

> I was thinking if the concern is AVC messages in the logs, wouldn't they also happen for `permissive`?

I think the main concern is being compatible with the official ISO.

**Comment by teg** [comment]
We seem to have CS9 issues again. I did not dig deeply, so please check first, but I would be ok with ignoring CS9 as a matter of principle if we have reason to believe the bug is in the compose.

We really need to test against snapshots 👼 

cc @ochosi

**Comment by gicmo** [comment]
> We seem to have CS9 issues again. I did not dig deeply, so please check first, but I would be ok with ignoring CS9 as a matter of principle if we have reason to believe the bug is in the compose.

Something is up with libvirt and UEFI:
`ERROR    Libvirt version does not support UEFI.`

**Comment by achilleas-k** [comment]
It's the bug flavour of the hour :roll_eyes:

**Comment by gicmo** [comment]
GREEN, but I merged the `C.UTF-8` PR and it now collides I think. Needs a rebase on main.

**Comment by gicmo** [comment]
I have rebased it on main and re-ran `./tools/test-case-generators/generate-test-cases --store /tmp/store --output test/data/manifests --distro {centos-9,rhel-90} --arch {aarch64,x86_64} --image-types edge-simplified-installer --keep-image-info`

**Review by gicmo — APPROVED** [review]
💿

**Pull request merged**